### PR TITLE
XSOAR mirroring fetch-incidents bug fix

### PR DESCRIPTION
**Pull request by yaakovpraisler**
## Status
- [x] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: [link to the issue](https://github.com/demisto/etc/issues/47633)

## Description
Fixed an issue where the `fetch-incidents` command missed some incidents.

## Minimum version of Cortex XSOAR
- [x] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0
- [ ] 6.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [x] No

## Must have
- [x] Tests
- [ ] Documentation

### CODE REVIEW

File: Packs/XSOARmirroring/Integrations/XSOARmirroring/XSOARmirroring.py [hunk 1]
```diff
@@ -332,18 +335,15 @@ def fetch_incidents(client: Client, max_results: int, last_run: Dict[str, int],
...
-        incident_created_time = arg_to_timestamp(
-            arg=incident.get('created'),  # type: ignore
-            arg_name='created',
-            required=True
-        )
+        incident_created_time = dateparser.parse(incident.get('created'))  # type: ignore
 
         # Update last run and add incident if the incident is newer than last fetch
-        if incident_created_time > latest_created_time:
+        if incident_created_time > latest_created_time:  # type: ignore
```
Comment (daryakoval) [inline Packs/XSOARmirroring/Integrations/XSOARmirroring/XSOARmirroring.py:341]: Please add the specific error name that you ignoring like here:

```suggestion
        if incident_created_time > latest_created_time:  # type: ignore[operator]
```

File: Packs/XSOARmirroring/Integrations/XSOARmirroring/XSOARmirroring.py [hunk 2]
```diff
@@ -734,11 +730,7 @@ def main() -> None:
     verify_certificate = not demisto.params().get('insecure', False)
 
     # How much time before the first fetch to retrieve incidents
-    first_fetch_time = arg_to_timestamp(
-        arg=demisto.params().get('first_fetch', '3 days'),
-        arg_name='First fetch time',
-        required=True
-    )
+    first_fetch_time = dateparser.parse(demisto.params().get('first_fetch', '3 days')).strftime(XSOAR_DATE_FORMAT)  # type: ignore
```
Comment (daryakoval) [inline Packs/XSOARmirroring/Integrations/XSOARmirroring/XSOARmirroring.py:733]:
```suggestion
    first_fetch_time = dateparser.parse(demisto.params().get('first_fetch', '3 days')).strftime(XSOAR_DATE_FORMAT)  # type: ignore[union-attr]
```

**Review by daryakoval — COMMENTED** [review]
Where is the link to failed test that reproduces the bug?

File: Packs/XSOARmirroring/Integrations/XSOARmirroring/XSOARmirroring.py [hunk 3]
```diff
@@ -270,27 +278,22 @@ def fetch_incidents(client: Client, max_results: int, last_run: Dict[str, int],
...
-    last_fetch: int = last_run.get('last_fetch', -1)
-    if last_fetch == -1:
+    last_fetch = last_run.get('last_fetch')
+    if not last_fetch:
         last_fetch = first_fetch_time  # type: ignore
-    else:
-        last_fetch = int(last_fetch)
 
-    latest_created_time = cast(int, last_fetch)
+    latest_created_time = dateparser.parse(last_fetch)
```
[the remark this answers is not in the document]
Comment (yaakovpraisler) [reply Packs/XSOARmirroring/Integrations/XSOARmirroring/XSOARmirroring.py:291]: You're probably right. Pushed a fix.

**Review by yaakovpraisler — COMMENTED** [review]
1. I'm not sure if I can add a test to reproduce the bug. Please correct me if I'm wrong.
2. Done.

File: Packs/XSOARmirroring/ReleaseNotes/2_0_6.md [hunk 4]
```diff
@@ -0,0 +1,4 @@
+#### Integrations
+##### XSOAR Mirroring
+- Fixed a bug where the **fetch-incidents** command missed some incidents.
```
Comment (dantavori) [inline Packs/XSOARmirroring/ReleaseNotes/2_0_6.md:3]:
```suggestion
- Fixed a bug where the ***fetch-incidents*** command missed some incidents.
```

**Review by dantavori — APPROVED** [review]

**Pull request merged**